### PR TITLE
OPTIM get_batch traffic when enable context-parallel

### PR DESCRIPTION
**Pull request by jingxu9x**
we can split batch's sequence-length before broadcast in tp_group, which can save time in get_batch

### CODE REVIEW

**Comment by uygnef** [comment]
Hi [Superkeyv](https://github.com/Superkeyv),

I encountered a similar issue while training with a sequence length of 32k on 8*H800 GPUs. At certain steps, the TP broadcast takes an extremely long time (about 10 seconds). This is quite unusual because the total data size is only 2 * 16k * 4 bytes (batch size * sequence length * 4 bytes, and we do not use an attention mask). In theory, this should only take a few us. Do you have any idea why this might be happening?

**Comment by jingxu9x** [comment]
@uygnef 
Can you explain in detail what the certain step did? do eval?

**Comment by uygnef** [comment]
I figured it out—the latency is due to the dataset server being under high load. This causes some steps to slow down. Setting a larger prefetch_factor should help fix it.
